### DISCUSSION TITLE
createAsyncStoragePersistor

### DISCUSSION THREAD

**Discussion by mtbaqer** [TanStack/query#2261] [Q&A]
Since both `LocalStorage` and `AsyncStorage` (for React Native) have a similar API interface, is there a reason why we don't have a `createAsyncStoragePersistor`?

Is it because of some technical difficulties or simply nobody got to it yet?

**Comment by TkDodo** [comment]
nobody got to it yet.

**Comment by dan-lee** [comment]
I just created a package that does it: https://github.com/dan-lee/react-query-native-persist
I'd also be happy to contribute it directly to `react-query`.

Not necessarily you need to use async-storage, you can provide an adapter for any storage backend. It provides two out of the box:
- `@react-native-async-storage/async-storage`
- `react-native-mmkv-storage`

Please report back if that fits your needs

**Reply by hirbod** [reply]
@dan-lee could you maybe extend your library to also support `react-native-mmkv`
https://github.com/mrousavy/react-native-mmkv

You already have mmkv storage support but for a different library.

**Reply by Balasnest** [reply]
Yeah. Much needed

**Reply by franferns** [reply]
@hirbod and @Balasnest where you able to get the **react-native-mmkv** to working with react query ?

**Reply by jensneuber** [reply]
Since this answer is missing here:

[mrousavy/react-native-mmkv@main/docs/WRAPPER_REACT_QUERY.md](https://github.com/mrousavy/react-native-mmkv/blob/main/docs/WRAPPER_REACT_QUERY.md)
